src/Razor/benchmarks/Microsoft.AspNetCore.Razor.Microbenchmarks/TextDifferencing/SourceTextDifferBenchmark.cs

```diff
@@ -1,58 +1,58 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-#nullable disable
-
 using System;
-using System.IO;
 using System.Linq;
 using BenchmarkDotNet.Attributes;
-using Microsoft.AspNetCore.Razor.LanguageServer;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.AspNetCore.Razor.TextDifferencing;
 using Microsoft.CodeAnalysis.Text;
 
-namespace Microsoft.AspNetCore.Razor.Microbenchmarks.Serialization;
+namespace Microsoft.AspNetCore.Razor.Microbenchmarks.TextDifferencing;
 
 public class SourceTextDifferBenchmark
 {
-    private readonly SourceText _largeFileOriginal;
-    private readonly SourceText _largeFileMinimalChanges;
-    private readonly SourceText _largeFileSignificantChanges;
+    private SourceText? _largeFileOriginal;
+    private SourceText? _largeFileMinimalChanges;
+    private SourceText? _largeFileSignificantChanges;
 
-    public SourceTextDifferBenchmark()
+    [GlobalSetup]
+    public void GlobalSetup()
     {
-        var current = new DirectoryInfo(AppContext.BaseDirectory);
-        while (current != null && !File.Exists(Path.Combine(current.FullName, "MSN.cshtml")))
-        {
-            current = current.Parent;
-        }
-
-        var largeFilePath = Path.Combine(current.FullName, "MSN.cshtml");
-        var largeFileText = File.ReadAllText(largeFilePath);
+        var largeFileText = Resources.GetResourceText("MSN.cshtml");
 
         _largeFileOriginal = SourceText.From(largeFileText);
 
         var changedText = largeFileText.Insert(100, "<");
         _largeFileMinimalChanges = SourceText.From(changedText);
 
-        changedText = largeFileText.Substring(largeFileText.Length / 2).Reverse().ToString();
```

Comment:
This is _not_ the right way to reverse a string or a `SourceText`. `Reverse()` is just an extension method that returns an `IEnumerable<char>`. Calling `ToString()` on the result doesn't foreach the enumerator and return a string, it returns the type name of the enumerable instance. Clearly, that isn't what the author of this benchmark intended, so I've gone ahead and produced the correct string below.